### PR TITLE
[tests] Replica Promotion improvements

### PR DESCRIPTION
**Pull request by Akasurde**
* Change string check in replica install at domain level 0
* Use hostname instead of host object in logger

Fixes: https://pagure.io/freeipa/issue/7104

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

**Comment by felipevolpone** [comment]
This maybe related to PR #902

**Comment by Akasurde** [comment]
closing and re-opening for CI trigger.

**Comment by Akasurde** [comment]
Do I need to do anything for this PR ?

**Comment by stlaz** [comment]
For starters, it would be good to remove the [WIP] from the title of the PR to let us know that you finished working on it 😃

**Comment by stlaz** [comment]
I'll check it, did not notice the removal of the [WIP] sign

File: ipatests/test_integration/test_replica_promotion.py [hunk 1]
```diff
@@ -52,8 +52,7 @@ def test_promotion_disabled(self):
                 '--ip-address', client.ip]
         result = client.run_command(args, raiseonerr=False)
         assert_error(result,
-                     'You must provide a file generated by ipa-replica-prepare'
-                     ' to create a replica when the domain is at level 0', 1)
+                     'You used the wrong mechanism to install a replica in domain level 0', 1)
```
Comment (stlaz) [inline ipatests/test_integration/test_replica_promotion.py]: Have you tested this? I am pretty sure that the text returned would be much longer.

Comment (felipevolpone) [inline ipatests/test_integration/test_replica_promotion.py]: @stlaz the `assert_error` method checks if the `'You used the wrong mechanism...'` string is `in` `resutl.stderr_text`

Comment (stlaz) [inline ipatests/test_integration/test_replica_promotion.py]: Ah, right, thanks, did not check the `assert_error` function.

**Comment by Akasurde** [comment]
bump for review/merge.

**Comment by stlaz** [comment]
I was testing this on F26 in my local virt machines, but forgot F26 is broken in the Cloud distribution and did not have time for this since. I'll eventually get to it.

**Comment by stlaz** [comment]
All the tests in the `TestRenewalMaster` class fail because client seems to be already installed:
```
E           CalledProcessError: Command '['ipa-client-install', '-U', '--domain', 'ipa.test', '--realm', 'IPA.TEST', '-p', 'admin', '-w', 'Secret123', '--server', 'master.ipa.test']' returned non-zero exit status 3
```
is the output from the tests and http://termbin.com/bhd4 can be found in `/var/log/ipaclient-install.log`

Note that 
```
test_integration/test_replica_promotion.py::TestKRAInstall::test_kra_install_without_replica_file
test_integration/test_replica_promotion.py::TestCAInstall::test_ca_install_without_replica_file
test_integration/test_replica_promotion.py::TestReplicaManageCommands::test_replica_manage_commands
test_integration/test_replica_promotion.py::TestProhibitReplicaUninstallation::test_replica_uninstallation_prohibited
```
were skipped in this run.

**Comment by stlaz** [comment]
You'll need to rebase on the current master for the tests to pass.

Not sure about those tests I mentioned, though, I believe I tested this with the PR rebased already. I'd need an explanation of why those failed.

**Comment by stlaz** [comment]
@Akasurde ?

**Comment by stlaz** [comment]
@Akasurde I'll close this PR week away from now if I don't see any activity.

**Comment by stlaz** [comment]
@Akasurde one thing you can do after the rebase is to create jobs in our upstream jenkins showing that the test were failing before your patch and are green after it. This is what we're doing with @Rezney and is enough for me to ACK this PR.

**Comment by felipevolpone** [comment]
@Akasurde can you please rebase again you PR to get the necessary changes to have travis green?

**Comment by stlaz** [comment]
Inactive, closing.

**Pull request closed without merging**